lib/ts/recipe/session/with-jwt/sessionClass.ts

```diff
@@ -55,8 +56,78 @@ export default class SessionClassWithJWT implements SessionContainerInterface {
     getExpiry = (userContext?: any): Promise<number> => {
         return this.originalSessionClass.getExpiry(userContext);
     };
+    getSessionClaimPayload = (userContext?: any): SessionClaimPayloadType => {
+        return this.originalSessionClass.getSessionClaimPayload(userContext);
+    };
+
+    // These are re-implemented here, because they can update the access token payload
+    updateClaim = async (claim: SessionClaim<any>, userContext?: any): Promise<void> => {
+        await this.updateClaims([claim], userContext);
+    };
+
+    updateClaims = async (claims: SessionClaim<any>[], userContext?: any): Promise<void> => {
+        const origSessionClaimPayloadJSON = JSON.stringify(this.getSessionClaimPayload());
+        const origAccessTokenPayloadJSON = JSON.stringify(this.getAccessTokenPayload());
+
+        let newSessionClaimPayload = this.getSessionClaimPayload();
+        let newAccessTokenPayload = this.getAccessTokenPayload();
+        for (const claim of claims) {
+            const value = await claim.fetch(this.getUserId(), userContext);
+            if (value !== undefined) {
+                newSessionClaimPayload = claim.addToPayload(this.getSessionClaimPayload(), value, userContext);
+            }
+            if (claim.updateAccessTokenPayload) {
+                newAccessTokenPayload = claim.updateAccessTokenPayload(
+                    this.getAccessTokenPayload(),
+                    value,
+                    userContext
+                );
+            }
+        }
+        const sessionClaimPayloadUpdate =
+            JSON.stringify(newSessionClaimPayload) !== origSessionClaimPayloadJSON ? newSessionClaimPayload : undefined;
+        const accessTokenPayloadUpdate =
+            JSON.stringify(newAccessTokenPayload) !== origAccessTokenPayloadJSON ? newAccessTokenPayload : undefined;
+        if (accessTokenPayloadUpdate !== undefined || sessionClaimPayloadUpdate !== undefined) {
+            await this.regenerateToken(accessTokenPayloadUpdate, sessionClaimPayloadUpdate, userContext);
+        }
```

Comment:
I feel like the updating the sessionClaims code can be refactored and moved to a place where all the functions can call it.